File: app/src/main/java/com/shrikantbadwaik/tmdb/data/di/ApplicationInjector.java
package com.shrikantbadwaik.tmdb.data.di;

import com.shrikantbadwaik.tmdb.UpcomingMoviesApplication;
import com.shrikantbadwaik.tmdb.data.di.component.ApplicationComponent;
import com.shrikantbadwaik.tmdb.data.di.component.DaggerApplicationComponent;

public class ApplicationInjector {
    public static void inject(UpcomingMoviesApplication upcomingMoviesApplication) {
        ApplicationComponent applicationComponent = DaggerApplicationComponent.builder()
                .application(upcomingMoviesApplication)
                .build();
        applicationComponent.inject(upcomingMoviesApplication);
    }
}

File: app/src/main/java/com/shrikantbadwaik/tmdb/data/di/ApplicationInjector.kt
package com.shrikantbadwaik.tmdb.data.di

import com.shrikantbadwaik.tmdb.UpcomingMoviesApplication
import com.shrikantbadwaik.tmdb.data.di.component.DaggerApplicationComponent

object ApplicationInjector {
    fun inject(upcomingMoviesApplication: UpcomingMoviesApplication) {
        val applicationComponent = DaggerApplicationComponent.builder().application(upcomingMoviesApplication).build()
        applicationComponent.inject(upcomingMoviesApplication)
    }
}